### PR TITLE
3 - error handling

### PR DESCRIPTION
**Pull request by WesJourdan**
- Update server controllers to return errors in a way that the front end can deal with them.
- Update web stores/services to handle server errors while passing the error messages through to components.

ALSO:
- Log user in on successful register.
- Reconfigure redux store setup to clear store on logout.

NOTE: I didn't touch `sessionsController.mobileLogin` yet. Will update it once this is approved.

### CODE REVIEW

File: server/global/handlers/authHandler.js [hunk 1]
```diff
@@ -8,7 +9,11 @@ module.exports = {
     // check by passport session
     if(!req.isAuthenticated()) {
       console.log("UNAUTHORIZED");
-      res.status(403).send("UNAUTHORIZED - NOT LOGGED IN");
+      // must use `.json` because the front end expects to parse the response using response.json()
+      res.status(401).json("You must be logged in to perform this action");
+      // both below are equivalent to above as far as the front end is concerned
+      // res.status(401).send(JSON.stringify("You must be logged in to perform this action"))
+      // throw new YoteError("You must be logged in to perform this action", 401)
```
Comment (grantfowler) [inline server/global/handlers/authHandler.js:16]: i actually kind of like the YoteError

Comment (grantfowler) [reply server/global/handlers/authHandler.js:16]: although maybe thats better as an explicit "error" catch - this is more of a message?

File: server/resources/product/productController.js [hunk 2]
```diff
@@ -48,11 +55,15 @@ exports.updateSingleById = async (req, res) => {
...
 exports.deleteSingle = async (req, res) => {
-  const oldProduct = await Product.findById(req.params.id)
-  if(!oldProduct) {
-    throw new YoteError("Could not find matching Product", 404)
-  }
-  const deletedProduct = await oldProduct.remove()
+  const oldProduct = await Product.findById(req.params.id).catch(err => {
+    console.log(err)
+    throw new YoteError("Error finding Product", 404)
+  });;
```
Comment (grantfowler) [inline server/resources/product/productController.js:61]: doubled up

File: web/src/config/store.js [hunk 3]
```diff
@@ -16,4 +14,17 @@ export const initStore = (loggedInUser = null) => configureStore({
       , error: null
     }
   }
+})
+
+const rootReducer = (state, action) => {
+  // clear store on logout, also on login so any previous rejected queries are cleared out.
```
Comment (grantfowler) [inline web/src/config/store.js:20]: +1

**Review by grantfowler — APPROVED** [review]

File: server/resources/product/productController.js (hunk 2 above)
Comment (WesJourdan) [reply server/resources/product/productController.js:61]: I don't understand.

**Pull request merged**